File: 14-longest-common-prefix/longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
            int n = strs.length;
        if(strs == null || n == 0){
            return "";
        }
    String a = strs[0];
        for(int i = 1;i<n;i++){
            while(strs[i].indexOf(a) != 0){
        a = a.substring(0,a.length() - 1);
            if(a.isEmpty()){
                return "";
            }
        }
    }
        return a;
    }
}

File: 14-longest-common-prefix/longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if(len(strs) == 0):
            return ""
        a = strs[0]
        for i in range(1,len(strs)):
            while not strs[i].startswith(a):
                a  = a[:-1]
        return a

        